pkg/upgrade/upgrades/v24_3_sql_instances_add_draining.go

```diff
@@ -9,35 +9,24 @@ import (
 	"context"
 
 	"github.com/cockroachdb/cockroach/pkg/clusterversion"
-	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descs"
-	"github.com/cockroachdb/cockroach/pkg/sql/catalog/systemschema"
 	"github.com/cockroachdb/cockroach/pkg/upgrade"
-	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
 	"github.com/cockroachdb/errors"
 )
 
 // sqlInstancesAddDrainingMigration adds a new column `is_draining` to the
 // system.sql_instances table.
 func sqlInstancesAddDrainingMigration(
-	ctx context.Context, cs clusterversion.ClusterVersion, deps upgrade.TenantDeps,
+	ctx context.Context, _ clusterversion.ClusterVersion, deps upgrade.TenantDeps,
 ) error {
-	finalDescriptor := systemschema.SQLInstancesTable()
-	// Replace the stored descriptor with the bootstrap descriptor.
+	// Update the sql_instance table to add the is_draining column.
 	err := deps.DB.DescsTxn(ctx, func(ctx context.Context, txn descs.Txn) error {
-		expectedDesc := finalDescriptor.TableDesc()
-		mutableDesc, err := txn.Descriptors().MutableByID(txn.KV()).Table(ctx, expectedDesc.GetID())
-		if err != nil {
-			return err
-		}
-		version := mutableDesc.Version
-		mutableDesc.TableDescriptor = *protoutil.Clone(expectedDesc).(*descpb.TableDescriptor)
-		mutableDesc.Version = version
-		return txn.Descriptors().WriteDesc(ctx, false, mutableDesc, txn.KV())
+		_, err := txn.Exec(ctx, "add-draining-column", txn.KV(),
+			`ALTER TABLE system.sql_instances ADD COLUMN IF NOT EXISTS is_draining BOOL NULL FAMILY "primary"`)
```

Comment:
Question: We have a helper function called migrateTable that does a lot of work that I sort of assume the schema changer already does for us. I suppose we only need that when we want to be idempotent based on some property other than the existence of a column.

TIL: I'm rather surprised we had to specify family here since you wouldn't need to for an ordinary table. But it looks like we have a special case in allocateColumnFamilyIDs for the system db.

Reply:
Done.

Its a bit safer to use that, so I switched the code over to it. The advantage of that one is that it waits for pending schema changes too (in case something crashes).
<!-- Reviewable comment -OC8snWb3cedCnrZ46T3:bhfnk5w -->
<!-- Sent from Reviewable.io -->